packages/protocol/contracts/governance/Governance.sol

```diff
@@ -527,6 +527,7 @@ contract Governance is
     nonReentrant
     returns (bool)
   {
+    require(queue.contains(proposalId), "cannot upvote a proposal not in the queue");
     dequeueProposalsIfReady();
```

Comment:
The orginal idea was to move`dequeueProposalsIfReady` downward of the function, but if we do that, we will get into the situation where we attempt to upvote a proposal whose time already passed the `dequeueFrequency`

The current code will remove that proposal from the queue early, making the above situation not possible in the first place

Reply:
Seem like the main issue raised by [#10091](https://github.com/celo-org/celo-monorepo/issues/10091) was that the proposal ID could change during execution, causing it to fail. 

Couldn't that still happen even with this change?

Would it be worth removing `dequeueProposalsIfReady()` from the `upvote()` and instead have `require(now >= lastDequeue.add(dequeueFrequency),"Must first dequeue proposals")`

This would sometimes require an extra interaction by the user, but would resolve the "in-fly" proposalID changes.

Reply:
Yes that would still happen in this change. Technically we decided to leave the function as is, the check introduced in line 530 is there to allow the code to fail early if the queue does not contain the proposalID in the first place before doing any work. Also the check is still left in line 544 because after `dequeueProposalsIfReady();` is executed, the proposal could have left the queue as well.

interesting perspective to `dequeueProposalsIfReady()` entirely from the function, and allow the user side explicitly call it as against doing things implicitly.
wonder what else would be impacted with such a change - as it could be breaking..but i'm willing to explore that..will setup a small discussion.